src/languages/javascript.js

```diff
@@ -252,7 +252,7 @@ export default function(hljs) {
 
   const CLASS_REFERENCE = {
     relevance: 0,
-    match: /\b[A-Z][a-z]+([A-Z][a-z]+)*/,
+    match: /\b[A-Z][A-Za-z0-9_]+/,
```

Comment:
This is okey, although technically it should be  `*` and not `+` (otherwise `new X()` won't work). But I'm not sure if this breaks some other logic on HighlightJS.

Reply:
I think one letter class names is an edge case.  We really want to see the camel case pattern to trigger this heuristic.

Reply:
Let's focus more on the tests... at the very end I might take one last look and tweak this a variety of diff ways.

Reply:
I think we need BOTH... I played with this briefly.

`CRV` with zero context could be a constant (all CAPS is a constant) or a class name... Changing this rule as it is now incorrectly identifies constants as classes (or vice versa)... I think perhaps we need to handle both:

- `CRV` (constant)
- `CRV(` (class)

So we'd use the camel case rule (with zero other context) PLUS the newer new rule proposed here, but only if a look-ahead sees a `(`.  I think that would cover a lot of cases.  Thoughts?